### ISSUE TITLE
Packages for Linux / OSX / Windows

### ISSUE THREAD

**Issue by conradkleinespel** [conradkleinespel/rooster#6]
It would be cool to have packages for the main platforms people might use Rooster on. This would make it easier to install. People shouldn't need to know that Rooster runs on Rust or uses 3rd party libraries in order to use Rooster.

Here are options for some OSs:
- Linux (general): get a [PackageCloud](https://packagecloud.io/) account or https://bintray.com/account/pricing?tab=account&type=pricing,
- ArchLinux: create a PKGBUILD using the [Create a package](https://wiki.archlinux.org/index.php/Creating_packages) docs,
- OSX: write a [Homebrew formula](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Formula-Cookbook.md),
- Windows: https://chocolatey.org/ ?

Let us know what you think might be the best way to go about this :smiley:

**Comment by qmx** [comment]
@conradkleinespel I'm packaging [limonite](https://github.com/qmx/limonite) for Fedora/CentOS/RHEL using [tito](https://github.com/dgoodwin/tito). I can't still recommend it since it forces on us a tagging scheme that is ugly to say the least :)

Once I figure out how to make things saner I'll be more than happy on helping to package Rooster for Red Hat flavors of linux :)

**Comment by conradkleinespel** [comment]
@qmx Thanks for sharing [tito](https://github.com/dgoodwin/tito). I didn't know about this tool. I wouldn't like to Rooster to be bound to a certain versioning scheme by a 3rd party tool. But maybe tito could still work out for us if the version it wants us to use is the same as what we would have used anyway, right?

It would be amazing if you could contribute Red Hat packaging scripts (or whatever it is that is needed for Red Hat packages, sorry I don't know anything about that at the moment) :smiley:

**Comment by sumproxy** [comment]
On Windows (if we get there) it could be [this](https://chocolatey.org/) unless someone suggests a better alternative

**Comment by conradkleinespel** [comment]
@sumproxy You are right about Windows. I don't think Rooster even compiles on Windows.

**Comment by conradkleinespel** [comment]
@qmx @sumproxy I've added a fancy new install script that currently works on Ubuntu (14.04, 16.04, 16.10), Debian 8, CentOS 7, Fedora 25 and FreeBSD 11. It can be run with a oneliner:
```shell
curl -sSL 'https://raw.githubusercontent.com/conradkleinespel/rooster/master/install.sh' | sh
```

Feel free to let me know what you think.

Of course, this does not exclude making proper packages for each distribution / OS, so I'm leaving this issue open until we get to that.

**Comment by conradkleinespel** [comment]
OSX package will be available once #16 is fixed, for other OSes or specific BSD/Linux distros, we'll keep track of progress in separate issues so that packages can be worked on independently from each other
